mlrun/db/httpdb.py

```diff
@@ -140,6 +153,54 @@ def get_base_api_url(self, path: str, version: str = None) -> str:
         url = f"{self.base_url}/{path_prefix}/{path}"
         return url
 
+    def request_with_retry(self, method, url, **kwargs):
+        max_retries = (
+            HTTP_RETRY_AMOUNT
+            if config.httpdb.retry_api_call_on_exception == "enabled"
+            else 1
+        )
```

Comment:
So, this is confusing - this is not really how much retry we attempt, it's how many tries we do altogether. The first call always happens. The default retry count (3) actually means we do 2 retries in case of a failure...